.pre-commit-config.yaml

```diff
@@ -37,16 +37,10 @@ repos:
     args: [--line-length=120, --fail-on-change]
 
 - repo: https://github.com/astral-sh/ruff-pre-commit
-  rev: v0.5.0
+  rev: v0.7.3
   hooks:
   - id: ruff-format
-    exclude: &exclude_ruff >
-      (?x)^(
-        docs/source/topics/processes/include/snippets/functions/parse_docstring_expose_ipython.py|
-        docs/source/topics/processes/include/snippets/functions/signature_plain_python_call_illegal.py|
-      )$
   - id: ruff
-    exclude: *exclude_ruff
```

Comment:
This was excluding files docs/source/topics/processes/include/snippets/functions/parse_docstring_expose_ipython.py and docs/source/topics/processes/include/snippets/functions/signature_plain_python_call_illegal.py from linting and formatting.

I've moved this configuration to pyproject.toml which I think is a much better place for it. For example, you can now run ruff also manually outside of pre-commit and it will have the same configuration.

Reply:
Ah this is reusing a yaml anchor! thanks!

Reply:
What you said, yes :sweat_smile:

Reply:
FTR: I am refusing to google what is "yaml anchor" :-D